### PR TITLE
Feature/405534 sent to regulator notification

### PR DESCRIPTION
**Pull request by Ehsan-Hatami**

### CODE REVIEW

File: src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs [hunk 1]
```diff
@@ -358,12 +358,29 @@ public async Task<IActionResult> UpdateDetailsConfirmation()
...
-            UpdatedDatetime = DateTime.UtcNow
+            UpdatedDatetime = DateTime.Now
         };
 
         return View(nameof(UpdateDetailsConfirmation), model);
     }
 
+
+    [HttpGet]
+    [AllowAnonymous]
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs:369]: I don't think this should be anonymous - the user requires to be logged in, surely?

File: src/FrontendAccountManagement.Web/ViewModels/AccountManagement/DetailsChangeRequestedViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using FrontendAccountManagement.Web.Extensions;
+
+namespace FrontendAccountManagement.Web.ViewModels.AccountManagement
+{
+    public class DetailsChangeRequestedViewModel
+    {
+        public string Username { get; set; }
+        public DateTime UpdatedDatetime { get; set; }
+
+        public string GetFormattedChangeMessage()
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/ViewModels/AccountManagement/DetailsChangeRequestedViewModel.cs:10]: I think view models should just have properties on them. Any methods on them should be created as extension methods

File: src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs (hunk 1 above)
Comment (Ehsan-Hatami) [reply src/FrontendAccountManagement.Web/Controllers/AccountManagement/AccountManagementController.cs]: removed now

File: src/FrontendAccountManagement.Web/ViewModels/AccountManagement/DetailsChangeRequestedViewModel.cs (hunk 2 above)
Comment (Ehsan-Hatami) [reply src/FrontendAccountManagement.Web/ViewModels/AccountManagement/DetailsChangeRequestedViewModel.cs]: defined extension methods now

File: src/FrontendAccountManagement.Web/Extensions/UpdateDetailsConfirmationViewModelExtension.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using FrontendAccountManagement.Web.ViewModels.AccountManagement;
+
+namespace FrontendAccountManagement.Web.Extensions
+{
+    public static class UpdateDetailsConfirmationViewModelExtension
+    {
+        public static string GetFormattedChangeMessage(this UpdateDetailsConfirmationViewModel model)
```
Comment (NikPeters-Atos) [inline src/FrontendAccountManagement.Web/Extensions/UpdateDetailsConfirmationViewModelExtension.cs:7]: You seem to have the same code here. Either create a static class that both can call, or have a base view model that the extension method uses

Comment (Ehsan-Hatami) [reply src/FrontendAccountManagement.Web/Extensions/UpdateDetailsConfirmationViewModelExtension.cs]: made the change

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**